src/test/run-pass/backtrace-debuginfo.rs

```diff
@@ -75,7 +76,17 @@ fn dump_filelines(filelines: &[Pos]) {
         println!("{}:{}", basename, line);
     }
 }
-
+// Skip top frame for MSVC, because it sees the macro
+#[cfg(target_env = "msvc")]
+fn dump_filelines(filelines: &[Pos]) {
+    let mut iter = filelines.iter().rev();
+    let _ = iter.next();
+    for &(file, line) in iter {
+        // extract a basename
+        let basename = file.split(&['/', '\\'][..]).last().unwrap();
+        println!("{}:{}", basename, line);
+    }
+}
```

Comment:
Instead of duplicating the function, could this do something like:

```rust
// explanatory comment
let skip = if cfg!(target_env = "msvc") {1} else {0};
for &(file, line) in filelines.iter().rev().skip(skip) {
    // ...
}
```